src/Mixins/Expectation.php

```diff
@@ -844,6 +844,7 @@ public function toMatchSnapshot(string $message = ''): self
 
         $string = match (true) {
             is_string($this->value) => $this->value,
+            is_object($this->value) && method_exists($this->value, 'toSnapshot') => $this->value->toSnapshot(),
```

Comment:
Love it! Can you add a test on it?

Reply:
Test added in https://github.com/pestphp/pest/pull/1017/commits/2a54b5819d5583698ba53e87d3d0bfa33fa1b2c8